wordpress-mobile/WordPress-Android
# [Gutenberg] File Block upload failure handling

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2910

`gutenberg PR` https://github.com/WordPress/gutenberg/pull/27795
`gutenberg mobile PR` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2917

### Description
We are implementing this behavior because currently in WP Android we have no way to know which MIME types are supported by a site. This functionality is available on iOS already so the WP iOS app isn't allowed to get into this state as the user isn't able to upload an unsupported file type. Modifications will have to be done to the Consolidated Media Picker in the future to rectify this issue, but until then an interim solution will be utilized. 

The solution involves mechanisms for clearer error propagation to the File Block from the `GlueCode` that communicates with the editor. Currently, the user just sees the word "Error" when a file upload error occurs. Errors can vary in nature so now the specific native platforms can pass a more detailed message along if necessary. In this case,the logic here is passing an error message pertaining to unsupported file types being uploaded to a site. So when the error occurs the File Block will now show the custom message within the error UI container of the File Block.

To test:

1. Select a site that is on a free plan. 
2. Add a File Block. 
3. Add an audio file. 
4. Notice the error below. 

Before | After 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/103705583-4ad45a80-4f79-11eb-8b23-b6b585793a93.png" width="320"></kbd>  |  <kbd><img src="https://user-images.githubusercontent.com/1509205/103705588-4c9e1e00-4f79-11eb-985c-7cae4f33d1ae.png" width="320"></kbd>

PR submission checklist:
